### PR TITLE
Use a Python gitignore template, not a node one.

### PR DESCRIPTION
**Pull request by danielballan**
For historical reasons, this repo has a `.gitignore` based on a template for node projects rather than Python ones. This uses a Python template + some additions for our `.env` file with secrets, sphinx's build products and pytest's cache, which is a relatively new artifact of running pytest.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,110 +1,136 @@
...
-# Gradle:
-.idea/gradle.xml
-.idea/libraries
 
-# Mongo Explorer plugin:
-.idea/mongoSettings.xml
 
-## File-based project format:
-*.iws
+### Python template
```
Comment (Mr0grog) [inline .gitignore]: Looks like you got partially doubled content in here. Looks good otherwise!

**Comment by Mr0grog** [comment]
Also the oddball thing I often add is `scratch.*`, so I can keep a `scratch.md` or `scratch.py` file or whatever in the repo with a stash of random code bits and ideas and samples. Dunno if you’d want to include that or not.

File: .gitignore (hunk 1 above)
Comment (danielballan) [reply .gitignore]: Fixed

**Comment by Mr0grog** [comment]
👍 Merging!

**Pull request merged**